### PR TITLE
Better exponent treatment beyond SI prefixes

### LINKED ISSUE #1920: Inconistent numeric format on logarithmic axis labels
Hi all,

The following example shows inconsistent x-axis labels. 
Result: 10e+21, 100e+21, 1e+24, 10e+24, 100e+24, 1e+27
Expected: 1e+22, 1e+23, 1e+24, 1e+25, 1e+26, 1e+27

Can this be improved? Thanks!

```
<html>
<head> <script src="https://cdn.plot.ly/plotly-latest.min.js"></script> </head>
<body> 
<div id="TEST" class="plotly-graph-div"></div>
<script>
    window.PLOTLYENV=window.PLOTLYENV || {};
    window.PLOTLYENV.BASE_URL="https://plot.ly";
    Plotly.newPlot('TEST', [{"mode":"lines","y":[0.0,1.0],"type":"scatter","x":[1.0e22,1.0e27]}],
               {"xaxis":{"type":"log"},"margin":{"l":50,"b":50,"r":50,"t":60}}, {showLink: false});
</script>
</body>
</html>
```
CrossRef: [https://github.com/sglyon/PlotlyJS.jl/issues/132](https://github.com/sglyon/PlotlyJS.jl/issues/132)

### PR DESCRIPTION
**Pull request by alexcjohnson**
Fixes #1920 

In addition to no longer rounding the exponent to a factor of 3 when you get past the SI prefixes, I also changed it to `exponentformat: 'power'` there - ie "&times;10<sup>16</sup>" instead of "e+16" as we had before. I suspect using `exponentformat: 'e'` as the fallback was a decision made before we supported pseudo-html in tick labels.

screen shot 2017-08-04 at 5 30 13 pm

cc @etpinard @AndiMD

### CODE REVIEW

File: src/constants/numerical.js [hunk 1]
```diff
@@ -47,5 +47,11 @@ module.exports = {
...
      * Are two values nearly equal? Compare to 1PPM
      */
-    ALMOST_EQUAL: 1 - 1e-6
+    ALMOST_EQUAL: 1 - 1e-6,
+
+    /*
+     * not a number, but for displaying numbers: the "minus sign" symbol is
+     * wider than the regular ascii dash "-"
+     */
+    MINUS_SIGN: '\u2212'
```
Comment (alexcjohnson) [inline src/constants/numerical.js]: Currently this is only used in cartesian tick labels (and hover text) and this does get reused in other places... but I suspect there are places in other plot types where we use a regular dash that should be replaced by this one.

**Comment by rreusser** [comment]
It's not quite related, but this reminded me of #755 and led to #1932. 🎉  The test I added for that PR will break when this is merged and will need to be updated to fancy exponent formatting.

Otherwise this PR looks good to me. 💃

File: src/plots/cartesian/axes.js [hunk 2]
```diff
@@ -1359,6 +1361,21 @@ function formatLinear(ax, out, hover, extraPrecision, hideexp) {
...
+}
+
+// are we beyond the range of common SI prefixes?
+// 10^-16 -> 1x10^-16
+// 10^-15 -> 1f
+// ...
+// 10^14 -> 100T
+// 10^15 -> 1x10^15
+// 10^16 -> 1x10^16
+function beyondSI(exponent) {
```
Comment (etpinard) [inline src/plots/cartesian/axes.js]: Beautiful.

**Comment by etpinard** [comment]
Solid work. Thanks!

**Pull request merged**